src/AccessibilityInsights.SharedUx/Controls/TestTabs/ColorContrast.xaml.cs

```diff
@@ -111,21 +112,49 @@ public void SetElement(ElementContext ec)
             }
         }
 
+        private static AnalyzerVersion SelectAnalyzerVersion(Bitmap bitmap)
+        {
+            int versionThresholdSize = ConfigurationManager.GetDefaultInstance().AppConfig.ColorContrastVersionThresholdSize;
+            return (bitmap.Width * bitmap.Height) > versionThresholdSize ? AnalyzerVersion.V1 : AnalyzerVersion.V2;
+        }
+
         private void RunAutoCCA(Bitmap bitmap)
         {
-            var bmc = new BitmapCollection(bitmap, new ColorContrastConfigBuilder().Build());
-            var result = bmc.RunColorContrastCalculation();
-            var pair = result.MostLikelyColorPair;
+            AnalyzerVersion analyzerVersion = SelectAnalyzerVersion(bitmap);
+            Stopwatch stopwatch = Stopwatch.StartNew();
```

Comment:
Is the stopwatch just for telemetry? Can we only run it if telemetry is enabled?

Reply:
It _is_ purely for telemetry, but we don't have a typical pattern of trying to gate `Stopwatch` creation on whether or not telemetry is enabled. If we wanted to add that, we could approach it holistically. Based on the [reference code](https://referencesource.microsoft.com/#System/services/monitoring/system/diagnosticts/Stopwatch.cs,135) for `Stopwatch`, though, the overhead of this class is very low